src/hevm/test/test.hs

```diff
@@ -1468,6 +1497,182 @@ instance Arbitrary RLPData where
          return $ RLPData $ List [r | RLPData r <- ls])
    ]
 
+instance Arbitrary Word128 where
+  arbitrary = liftM2 fromHiAndLo arbitrary arbitrary
+
+instance Arbitrary Word256 where
+  arbitrary = liftM2 fromHiAndLo arbitrary arbitrary
+
+instance Arbitrary W256 where
+  arbitrary = fmap W256 arbitrary
+
+instance Arbitrary (Expr Storage) where
+  arbitrary = sized genStorage
+
+instance Arbitrary (Expr EWord) where
+  arbitrary = sized genWord
+
+instance Arbitrary (Expr Byte) where
+  arbitrary = sized genByte
+
+instance Arbitrary (Expr Buf) where
+  arbitrary = sized genBuf
+
+genByte :: Int -> Gen (Expr Byte)
+genByte 0 = fmap LitByte arbitrary
+genByte sz = oneof
+  [ liftM2 IndexWord subWord subWord
+  , liftM2 ReadByte subWord subBuf
+  ]
+  where
+    subWord = genWord (sz `div` 2)
+    subBuf = genBuf (sz `div` 2)
+
+genNat :: Gen Int
+genNat = fmap fromIntegral (arbitrary :: Gen Natural)
+
+genName :: Gen Text
+genName = fmap T.pack $ listOf1 (oneof . (fmap pure) $ ['a'..'z'] <> ['A'..'Z'])
+
+genWord :: Int -> Gen (Expr EWord)
+genWord 0 = frequency
+  [ (1, fmap Lit arbitrary)
```

Comment:
word generation is biased to generate a `Lit` more often than other constructors. I think this better reflects the kind of expressions we actually build when executing EVM, and makes the simplifier more likely to fire, so the tests aren't just a no-op....